server/middleware/startNewJourney.ts

```diff
@@ -0,0 +1,11 @@
+import type { Request, Response } from 'express'
+import { v4 as uuidv4 } from 'uuid'
+
+export default function startNewJourney(replaceUrl: string) {
+  return (req: Request, res: Response): void => {
+    // This code generates and inserts a journeyId uuid/guid into the original URL. As all subsequent redirects are
+    // relative, this journeyId remains in the URL for the complete journey allowing it to be associated with
+    // per-journey session data. See populateJourney.ts for how this journeyId parameter is used
```

Comment:
Are all redirects or links relative? What would happen if any were full links, or links to external services like DPS?

Reply:
There were some absolute URLs (those that started with /) in appointments which I changed to relative ones. It's definitely the case that I/we need to retest the journeys following this to confirm we don't have any absolute URLs

Any full links or links to external services will not be affected. They don't automatically get the journeyId. That's only added when the journey starts and will only be maintained either by using relative links or explicitly including it in a URL